### PR TITLE
core: Block MBS Image upload

### PR DESCRIPTION
**Pull request by ArtiomDivak**
Uploading image into MBS will be block inside TransferDiskImageCommand validation

Signed-off-by: Artiom Divak <adivak@redhat.com>

### CODE REVIEW

**Comment by bennyz** [comment]
Suggestion use a validation similar to https://github.com/oVirt/ovirt-engine/blob/513d01627450fbd49d38b27959b7d198b773bccd/backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/VmCommand.java#L619

And once transfer is supported (shouldn't be too hard to do) we can just enable it in `ManagedBlockStorageDomainValidator`

Regarding commit message and PR description: both upload and download are blocked, so I'd rephrase it to "transfer"

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java [hunk 1]
```diff
@@ -45,19 +45,7 @@
 import org.ovirt.engine.core.common.businessentities.VM;
 import org.ovirt.engine.core.common.businessentities.VmBackup;
 import org.ovirt.engine.core.common.businessentities.VmBackupPhase;
-import org.ovirt.engine.core.common.businessentities.storage.DiskBackupMode;
```
Comment (bennyz) [inline backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java:48]: not needed

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java [hunk 2]
```diff
@@ -187,6 +175,11 @@ protected boolean validateImageTransfer() {
         DiskImage diskImage = getDiskImage();
         DiskValidator diskValidator = getDiskValidator(diskImage);
         DiskImagesValidator diskImagesValidator = getDiskImagesValidator(diskImage);
+
+        if(diskImage.getDiskStorageType().name() == DiskStorageType.MANAGED_BLOCK_STORAGE.name()) {
```
[the remark this answers is not in the document]
Comment (ArtiomDivak) [reply backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java]: ACK

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java (hunk 1 above)
Comment (ArtiomDivak) [reply backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java:48]: ACK

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java [hunk 3]
```diff
@@ -193,6 +195,12 @@ protected boolean validateImageTransfer() {
                 validate(diskValidator.isDiskExists())
                 && validate(diskImagesValidator.diskImagesNotIllegal())
                 && validate(storageDomainValidator.isDomainExistAndActive());
+
+        if(diskImage.getDiskStorageType() == DiskStorageType.MANAGED_BLOCK_STORAGE
```
Comment (bennyz) [inline backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java:199]:
```suggestion
        if (diskImage.getDiskStorageType() == DiskStorageType.MANAGED_BLOCK_STORAGE
```

Comment (bennyz) [inline backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java:199]: doing this in validateImageTransfer will only work if the image is provided, which is always the case, but if support is somehow added to the UI it will not work, so perhaps it's good to also check just the storage domain in `validateCreateImage`

**Review by barpavel — COMMENTED** [review]
Please add `Bug-Url:` to the commit message after `Signed-off-by`, and also add `Bug-Url:` to the PR description.
`Signed-off-by` in the PR description IMHO is redundant, up to you.

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java [hunk 4]
```diff
@@ -165,6 +167,10 @@ protected boolean validateCreateImage() {
         ActionReturnValue returnValue = CommandHelper.validate(ActionType.AddDisk, getAddDiskParameters(),
```
Comment (bennyz) [inline backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java:167]: Should be something like this, no?
```java
StorageDomain storageDomain = storageDomainDao.get(getParameters().getStorageDomainId());
if (!isSupportedByManagedBlockStorageDomain(storageDomain)) {
    ...
}
      
ActionReturnValue returnValue = CommandHelper.validate(ActionType.AddDisk, getAddDiskParameters(),
...
```

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java [hunk 5]
```diff
@@ -162,9 +164,16 @@ protected void setActionMessageParameters() {
     }
 
     protected boolean validateCreateImage() {
+
```
Comment (bennyz) [inline backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java:167]: not needed

**Review by sleviim — CHANGES_REQUESTED** [review]

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java [hunk 6]
```diff
@@ -193,6 +199,12 @@ protected boolean validateImageTransfer() {
                 validate(diskValidator.isDiskExists())
```
[the remark this answers is not in the document]
Comment (ArtiomDivak) [reply backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java:202]: @bennyz can you please tell us what you think about this?

[the remark this answers is not in the document]
Comment (bennyz) [reply backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java:200]: doesn't matter to me, but it can be done separately

File: backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java [hunk 7]
```diff
@@ -162,6 +164,11 @@ protected void setActionMessageParameters() {
     }
 
     protected boolean validateCreateImage() {
+        StorageDomain storageDomain = storageDomainDao.get(getParameters().getStorageDomainId());
+        if (!isSupportedByManagedBlockStorageDomain(storageDomain)) {
+            return failValidation(EngineMessage.ACTION_TYPE_FAILED_UNSUPPORTED_ACTION_FOR_MANAGED_BLOCK_STORAGE_TYPE);
```
Comment (ahadas) [inline backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/disk/image/TransferDiskImageCommand.java:169]: I think I commented on that part within the other PR but this is the right place - shouldn't ACTION_TYPE_FAILED_UNSUPPORTED_ACTION_FOR_MANAGED_BLOCK_STORAGE_TYPE be added to the errors by the call to `isSupportedByManagedBlockStorageDomain` above?
